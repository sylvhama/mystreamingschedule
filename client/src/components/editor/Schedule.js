import React from 'react';
import PropTypes from 'prop-types';
import ScheduleForm from './ScheduleForm';
import ScheduleList from './ScheduleList';
import {fetchOptions} from '../../helpers';

class Schedule extends React.Component {

  render() {
    return(
      <div>
        <ScheduleList loading={this.props.loading} /> 
        <br />
        <ScheduleForm loading={this.props.loading} /> 
      </div>
    );
  }
}

Schedule.propTypes = {
  loading: PropTypes.bool.isRequired,
  displayMsg: PropTypes.func.isRequired,
  setLoading: PropTypes.func.isRequired
}

export default Schedule;