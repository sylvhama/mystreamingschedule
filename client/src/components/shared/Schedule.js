import React from 'react';
import PropTypes from 'prop-types';
import {List, ListItem} from 'material-ui/List';
import Divider from 'material-ui/Divider';
import Subheader from 'material-ui/Subheader';
import Avatar from 'material-ui/Avatar';
import {formatTime} from '../../helpers';

const days = [
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
  'Sunday'
];

class Schedule extends React.Component {

  formatSchedules(schedules) {
    const week = [];
    schedules.forEach((schedule) => {
      schedule.days.forEach((day) => {
        if(!week[day-1]) week[day-1] = [];
        week[day-1].push(schedule);
      });
    });
    week.forEach((day) => {
      day.sort((a, b) => {
        if(a.startHour < b.startHour) return -1;
        if(a.startHour > b.startHour) return 1;
        if(a.startMin < b.startMin) return -1;
        if(a.startMin > b.startMin) return 1;
        return 0;
      });
    });
    return week;
  }

  renderSecondaryText(schedule) {
    return (
      <span>
        <strong>From {formatTime(schedule.startHour, schedule.startMin)} to {formatTime(schedule.endHour, schedule.endMin)}</strong>
        <br />
        {schedule.program.description}
      </span>
    )
  }

  renderList(week, i) {
    if(!week[i]) return;
    return(
      week[i].map((schedule) => {
        return (
          <ListItem
            key={schedule._id}
            leftAvatar={<Avatar style={{margin: 8}} size={24} backgroundColor={schedule.program.color} />}
            primaryText={schedule.program.name}
            secondaryText={this.renderSecondaryText(schedule)}
            disabled={true}
          />
        )
      })
    )
  }

  render() {
    const week = this.formatSchedules(this.props.schedules);
    return (
      <div>
        {days.map((day, i) => {
          return (
            <div key={day}>
              <List>
                <Subheader>{day}</Subheader>
                { this.renderList(week, i) }
              </List>
              <Divider />
            </div>
          )
        })}
      </div>
    );
  }
}

Schedule.propTypes = {
  loading: PropTypes.bool.isRequired,
  schedules: PropTypes.array.isRequired,
  editMode: PropTypes.bool,
  removeSchedule: PropTypes.func
}

export default Schedule;
