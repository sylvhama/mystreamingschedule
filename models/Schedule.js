const mongoose = require('mongoose');

mongoose.Promise = global.Promise;

const scheduleSchema = new mongoose.Schema({
  program: {
    type: mongoose.Schema.ObjectId,
    ref: 'Program',
    required: 'Please supply a program'
  },
  created: {
    type: Date,
    default: Date.now
  },
  start: {
    type: Number,
    required: 'Please supply a start time'
  },
  end: {
    type: Number,
    required: 'Please supply an end time'
  },
  days: {
    type: [Number],
    required: 'Please supply days'
  },
  active: {
    type: Boolean,
    default: true
  }
});

module.exports = mongoose.model('Schedule', scheduleSchema);
