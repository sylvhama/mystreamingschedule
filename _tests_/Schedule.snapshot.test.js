import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Schedule from '../client/src/components/shared/Schedule';
import renderer from 'react-test-renderer';
import React from 'react';

test('Schedule component renders the schedule correctly', () => {
  const schedules = [{"_id":"59478ae256407607fc3f84d0","author":"593d2e1f0b701a2ddce28213","program":{"_id":"59478aa556407607fc3f84cd","author":"593d2e1f0b701a2ddce28213","name":"Lunch Play","description":"Eat and Play!","color":"#8bc34a","__v":0,"created":"2017-06-19T08:26:13.878Z"},"startHour":12,"startMin":0,"endHour":16,"endMin":0,"__v":0,"days":[1,2,3,4,5],"created":"2017-06-19T08:27:14.160Z"},{"_id":"59478c6b56407607fc3f84d1","author":"593d2e1f0b701a2ddce28213","program":{"_id":"59478ab556407607fc3f84cf","author":"593d2e1f0b701a2ddce28213","name":"Retro Night","description":"Retrogaming discovery, mostly NES and Super NES.","color":"#3f51b5","__v":0,"created":"2017-06-19T08:26:29.414Z"},"startHour":20,"startMin":0,"endHour":0,"endMin":0,"__v":0,"days":[3,1],"created":"2017-06-19T08:33:47.845Z"},{"_id":"59478c8356407607fc3f84d2","author":"593d2e1f0b701a2ddce28213","program":{"_id":"59478aab56407607fc3f84ce","author":"593d2e1f0b701a2ddce28213","name":"Midnight Speed Run","description":"","color":"#ffeb3b","__v":0,"created":"2017-06-19T08:26:19.036Z"},"startHour":20,"startMin":0,"endHour":0,"endMin":0,"__v":0,"days":[2,4],"created":"2017-06-19T08:34:11.804Z"}],
        rendered = renderer.create(
    <MuiThemeProvider>
      <Schedule schedules={schedules} />
    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
