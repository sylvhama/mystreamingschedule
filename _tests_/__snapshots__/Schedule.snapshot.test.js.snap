// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule component renders the schedule correctly 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Monday
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8bc34a",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Lunch Play
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              12:00 pm
               to 
              04:00 pm
            </strong>
            <br />
            Eat and Play!
          </span>
        </div>
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3f51b5",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Retro Night
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              08:00 pm
               to 
              00:00 am
            </strong>
            <br />
            Retrogaming discovery, mostly NES and Super NES.
          </span>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Tuesday
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8bc34a",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Lunch Play
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              12:00 pm
               to 
              04:00 pm
            </strong>
            <br />
            Eat and Play!
          </span>
        </div>
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffeb3b",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Midnight Speed Run
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              08:00 pm
               to 
              00:00 am
            </strong>
            <br />
            
          </span>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Wednesday
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8bc34a",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Lunch Play
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              12:00 pm
               to 
              04:00 pm
            </strong>
            <br />
            Eat and Play!
          </span>
        </div>
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#3f51b5",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Retro Night
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              08:00 pm
               to 
              00:00 am
            </strong>
            <br />
            Retrogaming discovery, mostly NES and Super NES.
          </span>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Thursday
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8bc34a",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Lunch Play
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              12:00 pm
               to 
              04:00 pm
            </strong>
            <br />
            Eat and Play!
          </span>
        </div>
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffeb3b",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Midnight Speed Run
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              08:00 pm
               to 
              00:00 am
            </strong>
            <br />
            
          </span>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Friday
      </div>
      <div>
        <div
          onTouchEnd={[Function]}
          open={null}
          style={
            Object {
              "backgroundColor": null,
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 16,
              "lineHeight": "16px",
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 16,
              "paddingLeft": 72,
              "paddingRight": 16,
              "paddingTop": 20,
              "position": "relative",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }
        >
          <div
            className={undefined}
            size={24}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#8bc34a",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 12,
                "height": 24,
                "justifyContent": "center",
                "left": 16,
                "margin": 8,
                "muiPrepared": true,
                "position": "absolute",
                "top": 16,
                "userSelect": "none",
                "width": 24,
              }
            }
          />
          <div
            style={
              Object {
                "muiPrepared": true,
              }
            }
          >
            Lunch Play
          </div>
          <span
            style={
              Object {
                "WebkitBoxOrient": null,
                "WebkitLineClamp": null,
                "color": "rgba(0, 0, 0, 0.54)",
                "display": null,
                "fontSize": 14,
                "height": 16,
                "lineHeight": "16px",
                "margin": 0,
                "marginTop": 4,
                "muiPrepared": true,
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "whiteSpace": "normal",
              }
            }
          >
            <strong>
              From 
              12:00 pm
               to 
              04:00 pm
            </strong>
            <br />
            Eat and Play!
          </span>
        </div>
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Saturday
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
  <div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": "0px 0px 8px 0px",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 14,
            "fontWeight": 500,
            "lineHeight": "48px",
            "muiPrepared": true,
            "paddingLeft": 16,
            "width": "100%",
          }
        }
      >
        Sunday
      </div>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      }
    />
  </div>
</div>
`;
